### ISSUE TITLE
Enabled set to false results in errors

### ISSUE THREAD

**Issue by JamesTimms** [cloudposse/terraform-aws-vpc-peering-multi-account#19]
## Describe the Bug
When setting enabled to false you get errors in the `aws_vpc_peering_connection_options.accepter` and the `aws_vpc_peering_connection_options.requester`. The errors seem to be because these two resources are missing `count`.

## Expected Behavior
When I set enabled to false I expect the module to do nothing.

## Steps to Reproduce
1. Set enabled to false with other variables filled with empty strings
2. Run `terraform plan` (no errors should be shown)
3. Run `terraform apply` (see errors)

## Error Message
```
Error: error reading VPC Peering Connection (): InvalidVpcPeeringConnectionId.Malformed: Invalid id: "" (expecting "pcx-...")
        status code: 400, request id: c62fad7b-fc41-45d5-9fcb-6b97d7836638

  on .terraform/modules/vpc_peering_2/accepter.tf line 105, in resource "aws_vpc_peering_connection_options" "accepter":
 105: resource "aws_vpc_peering_connection_options" "accepter" {



Error: error reading VPC Peering Connection (): InvalidVpcPeeringConnectionId.Malformed: Invalid id: "" (expecting "pcx-...")
        status code: 400, request id: fc8948ec-92c9-4901-834d-9e6263055096

  on .terraform/modules/vpc_peering_2/requester.tf line 120, in resource "aws_vpc_peering_connection_options" "requester":
 120: resource "aws_vpc_peering_connection_options" "requester" {
```

## Environment (please complete the following information):

Anything that will help us triage the bug will help. Here are some ideas:
 - OS: Ubuntu 18.04.4
 - Code Version: 0.7.0 (but I see the missing count in master)

## Additional Context
I'm trying to loop through a list of VPC Peers but due to count in modules not being supported I've created three modules that I've set to `enabled=false` if the list of VPC Peers doesn't have a second or third element.

```
locals {
  # A dirty hack to support the below module hack
  vpc_peers = [for index in [0, 1, 2] : {
    peer_vpc_id    = (length(var.vpc_peers) > index) ? data.terraform_remote_state.aws_peer_vpcs[index].outputs.vpc_id : ""
    peer_region    = (length(var.vpc_peers) > index) ? data.terraform_remote_state.aws_peer_vpcs[index].outputs.region : ""
    peer_vpc_name  = (length(var.vpc_peers) > index) ? data.terraform_remote_state.aws_peer_vpcs[index].outputs.vpc_name : ""
    requester_role = (length(var.vpc_peers) > index) ? var.vpc_peers[index].requester_role : ""
    accepter_role  = (length(var.vpc_peers) > index) ? var.vpc_peers[index].accepter_role : ""
    enabled        = length(var.vpc_peers) > index
  }]
}

# A dirty hack as count and for_each is not supported for modules until 0.13 (currently 0.12)
module "vpc_peering_1" {
  source    = "git::https://github.com/cloudposse/terraform-aws-vpc-peering-multi-account.git?ref=0.7.0"
  namespace = terraform.workspace
  stage     = var.env
  name      = "${data.terraform_remote_state.this.outputs.vpc_name}-to-${local.vpc_peers[0].peer_vpc_name}"
  enabled   = local.vpc_peers[0].enabled

  requester_aws_assume_role_arn             = local.vpc_peers[0].requester_role
  requester_region                          = var.aws_region
  requester_vpc_id                          = data.terraform_remote_state.this.outputs.vpc_id
  requester_allow_remote_vpc_dns_resolution = true

  accepter_aws_assume_role_arn             = local.vpc_peers[0].accepter_role
  accepter_region                          = var.aws_region
  accepter_vpc_id                          = local.vpc_peers[0].peer_vpc_id
  accepter_allow_remote_vpc_dns_resolution = true
}

# A dirty hack as count and for_each is not supported for modules until 0.13 (currently 0.12)
module "vpc_peering_2" {
  source    = "git::https://github.com/cloudposse/terraform-aws-vpc-peering-multi-account.git?ref=0.7.0"
  namespace = terraform.workspace
  stage     = var.env
  name      = "${data.terraform_remote_state.this.outputs.vpc_name}-to-${local.vpc_peers[0].peer_vpc_name}"
  enabled   = local.vpc_peers[1].enabled

  requester_aws_assume_role_arn             = local.vpc_peers[1].requester_role
  requester_region                          = var.aws_region
  requester_vpc_id                          = data.terraform_remote_state.this.outputs.vpc_id
  requester_allow_remote_vpc_dns_resolution = true

  accepter_aws_assume_role_arn             = local.vpc_peers[1].accepter_role
  accepter_region                          = var.aws_region
  accepter_vpc_id                          = local.vpc_peers[1].peer_vpc_id
  accepter_allow_remote_vpc_dns_resolution = true
}
```

**Comment by JamesTimms** [comment]
If it's just the missing `count` I don't mind submitting a PR for this.

**Comment by JamesTimms** [comment]
I've found another related issue present only in Terraform 0.12.11+ caused by a bug fix to [#22846](https://github.com/hashicorp/terraform/issues/22846). As it's also caused by the same conditions as this issue I think it's worth treating it as related.

With my example above I haven't shown vpc_peering_3 but it is there and is set to enabled = false which leads to this error below in Terraform 0.12.11 (and 0.12.29).

```
module.vpc_peering_2.aws_route.requester[0]: Refreshing state... [id=r-rtb-0eb93b6ecf5c1979b4017255854]
module.vpc_peering_2.aws_route.requester[1]: Refreshing state... [id=r-rtb-0f3bdf75751185fc54017255854]

Error: Invalid index

  on .terraform/modules/vpc_peering_3/accepter.tf line 76, in locals:
  76:   accepter_aws_route_table_ids           = distinct(sort(data.aws_route_tables.accepter[0].ids))
    ----------------
    data.aws_route_tables.accepter is empty tuple

The given key does not identify an element in this collection value.

10:20:11: [ERROR] terraform plan failed with: 1
10:20:11: [ERROR] Failed to plan all modules.
```

**Comment by JamesTimms** [comment]
This is now fixed with PR #20
